tests/conftest.py

```diff
@@ -0,0 +1,113 @@
+import os
+
+import numpy as np
+import pandas as pd
+import pytest
+
+import audb
+import audeer
+import audformat
+import audiofile
+
+
+pytest.NAME = 'db'
+pytest.VERSION = '1.0.0'
+
+
+@pytest.fixture
+def db(tmpdir, scope='function'):
+    r"""Publish a test database.
+
+    The database will use ``pytest.NAME`` as name
+    and ``pytest.VERSION`` as version.
+    It will be published to a file-system repository
+    in a tmp folder.
+    ``audb.config.REPOSITORIES`` will be adjusted
+    to point to the repository
+    the database was published to.
+
+    """
+    db_path = audeer.mkdir(audeer.path(tmpdir, pytest.NAME))
+
+    db = audformat.Database(
+        name=pytest.NAME,
+        source='https://github.com/audeering/audbcards',
+        usage='unrestricted',
+        expires=None,
+        languages='eng',
+        description='Example database.',
+        author='H Wierstorf, C Geng, B E Abrougui',
+        organization='audEERING',
+        license=audformat.define.License.CC0_1_0,
+    )
+
+    # Misc table 'speaker'
+    db.schemes['age'] = audformat.Scheme(
+        'int',
+        minimum=0,
+        description='Age of speaker',
+    )
+    db.schemes['gender'] = audformat.Scheme(
+        'str',
+        labels=['female', 'male'],
+        description='Gender of speaker',
+    )
+    index = pd.Index(
+        [0, 1],
+        dtype='Int64',
+        name='speaker',
+    )
+    db['speaker'] = audformat.MiscTable(index)
+    db['speaker']['age'] = audformat.Column(scheme_id='age')
+    db['speaker']['age'].set([23, 49])
+    db['speaker']['gender'] = audformat.Column(scheme_id='gender')
+    db['speaker']['gender'].set(['female', 'male'])
+
+    # Table 'files'
+    db.schemes['speaker'] = audformat.Scheme(
+        'int',
+        labels='speaker',
+        description='Speaker IDs.',
+    )
+    index = audformat.filewise_index(['data/f0.wav', 'data/f1.wav'])
+    db['files'] = audformat.Table(index)
+    db['files']['speaker'] = audformat.Column(scheme_id='speaker')
+    db['files']['speaker'].set([0, 1])
+
+    # Table 'segments'
+    db.schemes['emotion'] = audformat.Scheme(
+        'str',
+        labels=['angry', 'happy', 'neutral'],
+        description='Emotional class.',
+    )
+    index = audformat.segmented_index(
+        files=['data/f0.wav', 'data/f0.wav', 'data/f1.wav', 'data/f1.wav'],
+        starts=[0, 1, 0, 1],
+        ends=[1, 2, 1, 2],
+    )
+    db['segments'] = audformat.Table(index)
```

Comment:
I understand that this fixture is supposed to work in  a "destructive fashion". For example assume that I need a db object without the segments scheme.  I would use the fixture and try to delete the segments scheme.
However I believe that there is no delete method (trying to delete schemes or set it to `None` results in `cannot delete` or `cannot set` ) errors.  How would such a problem be solved in the current fixture design?

Reply:
Of course one could manipulate `db.__dict__` but that would be very ugly.

Reply:
When you want to remove a scheme you can do this with, e.g.:
```python
del db.schemes['emotion']
```
But I guess you want to remove a table. This can be done with
```python
db.drop_tables(['segments'])
```
compare https://audeering.github.io/audformat/api/audformat.Database.html#drop-tables.

Or you might decide to not use the new `db` fixture and simply load parts of the database in the test, e.g.

```python
def test_something():
    db = audb.load(
        pytest.NAME,
        version=pytest.VERSION,
        tables=['files'],
        verbose=False,
)
```